waddrmgr/db.go

```diff
@@ -549,14 +581,14 @@ func putNumAccounts(tx walletdb.Tx, numAccounts uint32) error {
 // the common parts.
 func deserializeAddressRow(addressID, serializedAddress []byte) (*dbAddressRow, error) {
 	// The serialized address format is:
-	//   <addrType><account><addedTime><syncStatus><rawdata>
+	//   <addrType><account><addedTime><syncStatus><watchingOnly><rawdata>
```

Comment:
After thinking about it some more, I think having the extra field is more flexible.  However, at the moment there is some redundancy with the `adtWatching` type that doesn't quite feel right.

If we look at it from the perspective of what a user will need to do for watching-only purposes:

* Import BIP0032 public key chains
* Import "loose" public keys (those which are not a part of BIP0032 keychain)
* Import the hash160 associated with a pay-to-script-hash

Considering that, the first case is an `adtChain` with a watching-only flag, the second is an `adtImport` with a watching-only flag, and the third is an `adtScript` with a watching-only flag.